moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -319,40 +300,77 @@ void ServoCalcs::stop()
   }
 }
 
+void ServoCalcs::updateParams()
+{
+  if (servo_param_listener_->is_old(servo_params_))
+  {
+    auto params = servo_param_listener_->get_params();
+    if (params.override_velocity_scaling_factor != servo_params_.override_velocity_scaling_factor)
+    {
+      RCLCPP_INFO_STREAM(LOGGER, "override_velocity_scaling_factor changed to : "
+                                     << std::to_string(params.override_velocity_scaling_factor));
+    }
+
+    if (params.robot_link_command_frame != servo_params_.robot_link_command_frame)
+    {
+      if (current_state_->knowsFrameTransform(params.robot_link_command_frame))
+      {
+        RCLCPP_INFO_STREAM(LOGGER, "robot_link_command_frame changed to : " << params.robot_link_command_frame);
+      }
+      else
+      {
+        RCLCPP_ERROR_STREAM(LOGGER, "Failed to change robot_link_command_frame. Passed frame '"
+                                        << params.robot_link_command_frame
+                                        << "' is unknown, will keep using old command frame.");
+        // Replace frame in new param set with old frame value
+        // TODO : Is there a better behaviour here ?
+        params.robot_link_command_frame = servo_params_.robot_link_command_frame;
+      }
+    }
+    servo_params_ = params;
+  }
+}
+
 void ServoCalcs::mainCalcLoop()
 {
-  rclcpp::WallRate rate(1.0 / parameters_->publish_period);
+  rclcpp::WallRate rate(1.0 / servo_params_.publish_period);
 
   while (rclcpp::ok() && !stop_requested_)
   {
     // lock the input state mutex
     std::unique_lock<std::mutex> main_loop_lock(main_loop_mutex_);
 
     // low latency mode -- begin calculations as soon as a new command is received.
-    if (parameters_->low_latency_mode)
+    if (servo_params_.low_latency_mode)
     {
       input_cv_.wait(main_loop_lock, [this] { return (new_input_cmd_ || stop_requested_); });
     }
 
     // reset new_input_cmd_ flag
     new_input_cmd_ = false;
 
+    // Check if any parameters changed
+    if (servo_params_.enable_parameter_update)
+    {
+      updateParams();
+    }
```

Comment:
Shall we move this to the very top of the function? Because let's say we eventually have `low_latency_mode` also be responsive to this function, it wouldn't take effect til the next tick.